contrib_bots/lib/spotify.py

```diff
@@ -0,0 +1,124 @@
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'spotify':
+            return False
+        is_spotify = original_content.startswith('@spotify')
+
+        return is_spotify
+
+    def handle_message(self, message, client, state_handler):
+
+        def build_data(query_type, original_content):
+            if query_type = "song":
+                encoded_message = urllib.parse.quote(original_content[5:])
+                url = BASE_URL+encoded_message+"&type=track"+"&limit=10"
+                return urllib.request.urlopen(url).read()
+            elif query_type = "album":
+                encoded_message = urllib.parse.quote(original_content[6:])
+                url = BASE_URL+encoded_message+"&type=album"+"&limit=10"
+            elif query_type = "artist":
+                encoded_message = urllib.parse.quote(original_content[7:])
+                url = BASE_URL+encoded_message+"&type=artist"+"&limit=10"
+            return urllib.request.urlopen(url).read()
+
+        def artist_content(artist):
+            artist_name = artist['name']
+            artist_url = artist['external_urls']['spotify']
+            return [artist_name, artist_url]
+
+        def album_content(album):
+            album_name = album['name']
+            album_url = album['external_urls']['spotify']
+            artist_name = album['artists'][0]['name']
+            artist_url = album['artists'][0]['external_urls']['spotify']
+            return [album_name, album_url, artist_name, artist_url]
+
+        def track_content(track):
+            song_name = track['name']
+            song_url = track['external_urls']['spotify']
+            album_name = track['album']['name']
+            album_url = track['album']['external_urls']['spotify']
+            artist_name = track['artists'][0]['name']
+            artist_url = track['artists'][0]['external_urls']['spotify']
+            return [song_name, song_url, album_name, album_url, artist_name, artist_url]
+
+        BASE_URL = 'https://api.spotify.com/v1/search?q='
+        original_content = message['content']
+        original_content = original_content.replace('@spotify ', '')
+        new_content = ''
+
+        if original_content.split(' ', 1)[0] == "album":
+            new_content += " Here are the albums with \"%s\" in them: ".format(original_content[6:])
+            raw_data = build_data("album", original_content)
+            try:
+                raw_albums_list = json.loads(raw_data)['albums']['items']
+                if raw_albums_list != []:
+                    for album in raw_albums_list:
+                        content = album_content(album)
+                        new_content += "\n [%s](%s) by [%s](%s)\n".format(tuple(content))
+                else:
+                    new_content = "Sorry, we couldn't find anything with \"%s\".".format(original_content[6:])
```

Comment:
I checked it in Python interpreter - please fix all those issues and run the program. I'd love to see screenshots.